src/guide/diagrams-src/triggers.puml

```diff
@@ -0,0 +1,24 @@
+@startuml
+object event_trigger {
+  {Action<F: Filter>}
+}
+
+object time_trigger {
+  {Action<TimeEventFilter>}
+}
+object data_trigger {
+  {Action<DataEventFilter>}
+}
+object by_call_trigger {
+  {Action<ExecuteTriggerEventFilter>}
+}
+object pipeline_trigger {
+  {Action<PipelineEventFilter>}
+}
+
+event_trigger *-- time_trigger
+event_trigger *-- by_call_trigger
+event_trigger *-- data_trigger
+event_trigger *-- pipeline_trigger
```

Comment:
I'm not sure about this arrows. Looks like it means *composition*. And this is not true.

I.e. `time_trigger` is a ... type of `event_trigger`.

It's hard to reflect `Rust` type system and generics on UML that was designed for OOP languages. Maybe [this link](https://stackoverflow.com/a/4786989/10366988) can help more.

Or you can use just *association* here